core/arch/arm/kernel/generic_boot.c

```diff
@@ -669,6 +725,9 @@ static int add_res_mem_dt_node(void *fdt, const char *name, paddr_t pa,
 			return -1;
 	} else {
 		offs = fdt_path_offset(fdt, "/");
+		if (offs < 0)
+			return -1;
+		offs = add_dt_overlay_fragment(fdt, offs);
```

Comment:
How about adding a helper function for this pattern? (get "/" then add overlay fragment). It is the third time it is used in this file.